protocol/v2/ssv/runner/aggregator.go

```diff
@@ -98,16 +101,32 @@ func (r *AggregatorRunner) ProcessPreConsensus(logger *zap.Logger, signedMsg *sp
 		return errors.Wrap(err, "got pre-consensus quorum but it has invalid signatures")
 	}
 
+	// signer must be same for all messages, at least 1 message must be present (this is validated
+	// prior to ProcessPreConsensus call)
+	signer := signedMsg.Messages[0].Signer
 	duty := r.GetState().StartingDuty.(*spectypes.ValidatorDuty)
 
 	logger.Debug("🧩 got partial signature quorum",
-		zap.Any("signer", signedMsg.Messages[0].Signer), // TODO: always 1?
+		zap.Any("signer", signer),
 		fields.Slot(duty.Slot),
 	)
 
+	// this is the earliest in aggregator runner flow where we get to know whether we are meant
+	// to perform this aggregation duty or not
+	ok, err := isAggregator(duty.CommitteeLength, fullSig)
+	if err != nil {
+		return fmt.Errorf("check if validator is an aggregator: %w", err)
+	}
+	if !ok {
+		logger.Debug("aggregation duty won't be needed from this validator for this slot",
+			zap.Any("signer", signer),
+			fields.Slot(duty.Slot),
+		)
+		return nil
+	}
+
 	r.metrics.PauseDutyFullFlow()
 	// get block data
-	duty = r.GetState().StartingDuty.(*spectypes.ValidatorDuty)
```

Comment:
This line is essentially duplicated, looks like a typo (see couple of lines above it).